src/Console/Executor/DatabaseHyphenationExecutor.php

```diff
@@ -14,17 +14,27 @@
 
 class DatabaseHyphenationExecutor implements ExecutorInterface
 {
+    private ?string $word = null;
+
     public function __construct(
         private readonly WordRepository $wordRepository,
         private readonly RuleRepository $ruleRepository,
         private readonly DatabaseProcessor $processor,
-        private readonly string $word,
     ) {
 
     }
 
+    public function setWord(string $word): void
+    {
+        $this->word = $word;
+    }
+
     public function execute(): void
     {
+        if ($this->word === null) {
+            throw new \BadMethodCallException('Word should be set before calling execute() method');
```

Comment:
Throwing a `BadMethodCallException` might not be the most appropriate choice. While it's technically valid because the method is being called in an improper state (i.e., without setting the `$word` property first), the exception type isn't specific to the problem you're encountering: a missing or unset property.

My suggestions would be to use `LogicException` - as this can be considered a violation of internal program logic. The class is not used in a valid (logical) way. https://www.php.net/manual/en/class.logicexception.php